spec/models/manageiq/providers/embedded_ansible/automation_manager/job_spec.rb

```diff
@@ -83,14 +83,45 @@
           }
         end
 
+        before { Timecop.freeze }
+
+        let(:the_raw_plays) do
+          [
+            {
+              "id"         => 1,
+              "event"      => "playbook_on_play_start",
+              "failed"     => false,
+              "created"    => Time.current,
+              "event_data" => {"play" => "play1"}
+            },
+            {
+              "id"         => 2,
+              "event"      => "some_other_event",
+              "failed"     => false,
+              "created"    => Time.current + 5,
+              "event_data" => {"stdout" => "foo"}
+            },
+            {
+              "id"         => 3,
+              "event"      => "playbook_on_play_start",
+              "failed"     => true,
+              "created"    => Time.current + 10,
+              "event_data" => {"play" => "play2"}
+            }
+          ]
+        end
+
         it "syncs the job with the provider" do
+          fake_finish_time = the_raw_plays.last["created"] + 15
+          allow(subject).to receive(:finish_time).and_return(fake_finish_time)
+          expect(subject).to receive(:raw_stdout_json).and_return(the_raw_plays)
           subject.refresh_ems
 
           expect(subject).to have_attributes(
             :ems_ref     => subject.miq_task.id.to_s,
             :status      => subject.miq_task.state,
             :start_time  => subject.miq_task.started_on,
-            :finish_time => nil,
+            :finish_time => fake_finish_time,
```

Comment:
nice

Reply:
This should have been changed, but we will see.